File: software/matlab/instrumentos/Protek506.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% El multímetro Protek 506 se comunica con la compu a través de un puerto
%%% serie (DB9). El multímetro devuelve la lectura que muestra en pantalla 
%%% cada vez que desde el puerto serie recibe algo (puede ser un caracter 
%%% en blanco). En este ejemplo está seteado para leer temperatura con una 
%%% termocupla, por lo que el string que devuelve el multímetro es 
%%% 'TEMP 33,2'. De ahí el programa extrae el valor de temperatura y lo
%%% guarda en una variable
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
clear all;

%muestro la lista de equipos disponibles, y el string de inicializacion
out=busco_equipos

s2=serial('COM1','BaudRate',1200,'DataBits',7,'StopBits',2,'Parity','none','terminator','CR');
fopen(s2);
set(s2,'terminator','CR');


medicion=[];
temp=[];
tiempo=[];
tic
N=5;
for i=1:N;
medicion(i)=i
str=query(s2,'');
temperatura=sscanf(str,'TEMP %f')
temp(i)=temperatura;
tiempo(i)=toc;
end;

datos=[tiempo;temp ]';

fclose(s2);
delete(s2);
clear s2;


save('datatemp.txt','datos','-ascii')

plot(tiempo,temp)

File: software/python/instrumentos/Protek506.py
# -*- coding: utf-8 -*-
"""
Protek 506

 El multimetro Protek 506 se comunica con la compu a travis de un puerto
 serie (DB9). El multimetro devuelve la lectura que muestra en pantalla
 cada vez que desde el puerto serie recibe algo (puede ser un caracter
 en blanco). En este ejemplo esta seteado para leer temperatura con una
 termocupla, por lo que el string que devuelve el multimetro es
 'TEMP 33,2'. De ahi el programa extrae el valor de temperatura y lo
 guarda en una variable

Manual (web): https://github.com/hgrecco/labosdf-bin/raw/master/manuals/Agilent 34401.pdf
Manual (local): \\Srvlabos\manuales\Agilent\Agilent 34401.pdf
"""

import time

import visa

print(__doc__)

# Este string determina el intrumento que van a usar.
# Lo tienen que cambiar de acuerdo a lo que tengan conectado.
resource_name = 'ASRL1::INSTR'

rm = visa.ResourceManager()


mult = rm.open_resource(resource_name,
                        baudrate=1200, data_bits=7, stop_bits=visa.constants.StopBits.two,
                        parity=visa.constants.Parity.none,
                        write_termination='\n')

mediciones = []

for n in range(5):
    result = mult.query('')
    mediciones.append((time.time(), float(result[4:])))

print(mediciones)

mult.close()
